### PR TITLE
fix: update meilisearch indexing

### PR DESCRIPTION
**Pull request by eliasmpw**
Update script to validate objectID before trying to insert, also deleting obsolete indexes, and adding it to automation

### CODE REVIEW

File: scripts/meilisearchIndexer.ts [hunk 1]
```diff
@@ -43,27 +68,46 @@ const updateMeilisearchIndexes = async () => {
...
+    await waitTxComplete(client, res.taskUid);
+    console.log(`${innerDocs.length} entries updated!`);
+  }
+
+  // Delete old entries
+  const toDelete = Object.keys(oldEntries);
+  if (toDelete.length) {
+    const deleteRes = await client.index(indexName).deleteDocuments({ filter: `objectID IN [${toDelete.join(',')}]` });
+    console.log(`Enqueued the ${deleteRes.type} of ${toDelete.length} records`);
+    waitTxComplete(client, deleteRes.taskUid);
```
Comment (pyncz) [inline scripts/meilisearchIndexer.ts:95]: Not `await`? At least for the log below

File: scripts/meilisearchIndexer.ts [hunk 2]
```diff
@@ -43,27 +68,46 @@ const updateMeilisearchIndexes = async () => {
...
+
+  console.log('\nSuccessfully updated all docs! :)');
+};
+
+const waitTxComplete = async (client: MeiliSearch, taskUid: number) => {
+  let status;
 
-    console.log('Index successfully updated');
+  while (status !== 'succeeded') {
+    new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (pyncz) [inline scripts/meilisearchIndexer.ts:106]: I don't get this one. If it's to set the polling interval, should probably be also `await` (or just use `setInterval` in the first place)

**Review by pyncz — CHANGES_REQUESTED** [review]

File: scripts/meilisearchIndexer.ts (hunk 2 above)
Comment (pyncz) [reply scripts/meilisearchIndexer.ts:106]: Otherwise it seems it iterates again and again and again, as fast as js allows

Comment (eliasmpw) [reply scripts/meilisearchIndexer.ts]: You are right, at least it was awaiting line 108, but yeah, it would immediately retry after receiving a response

**Review by pyncz — APPROVED** [review]

**Pull request merged**